app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,26 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
+
+    public boolean theSameAs(T t){
+        return this.hashCode() == t.hashCode();
+    }
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.hashCode() == newItem.hashCode();
```

Comment:
Здесь нужно вызвать методы интерфейса `ListDiffInterface` у обоих аргументов функции:
```suggestion
        return oldItem.theSameAs(newItem);
```